src/utils/hooks/useGetFullApplicationStructure.tsx

```diff
@@ -134,13 +150,21 @@ const useGetFullApplicationStructure = (structure: FullStructure, firstRunValida
       evaluationParameters,
       true
     )
-    const results = await Promise.all([isEditable, isRequired, isVisible])
+    const isValid =
+      shouldProcessValidation || firstRunProcessValidation
+        ? evaluateExpressionWithFallBack(element.validationExpression, evaluationParameters, false)
+        : new Promise(() => responseObject[element.code]?.isValid)
+    setFirstRunProcessValidation(false)
+    const results = await Promise.all([isEditable, isRequired, isVisible, isValid])
+    if (shouldProcessValidation || firstRunProcessValidation) setLastValidationTimestamp(Date.now())
```

Comment:
just to be safe, maybe this should be down the bottom ?

Reply:
actually this is updating on eery question evaluated

Reply:
Yeah true, have moved it (and `setFirstRunProcessValidation`) back up to the same place as the other `set...` calls.

0436e17